### ISSUE TITLE
500 response with no body raises uninformative OAuth2::Error

### ISSUE THREAD

**Issue by ginjo** [ruby-oauth/oauth2#406]
Hi, thanks for this very useful gem!

I have an issue where I'm targeting an OAuth2 server (Slack) that is returning a '500 hphp_invoke' error with an empty body. This might be a bug in the server or in the data I'm sending. But the OAuth2 error raised is lacking in helpful information.

```ruby
irb(main):003:0> response = access_token.get('/api/auth.test')
OAuth2::Error: 
	from /usr/local/bundle/gems/oauth2-1.4.0/lib/oauth2/client.rb:119:in `request'
	from /usr/local/bundle/gems/oauth2-1.4.0/lib/oauth2/access_token.rb:107:in `request'
	from /usr/local/bundle/gems/oauth2-1.4.0/lib/oauth2/access_token.rb:114:in `get'
	from (irb):3

# that's all folks
```

To find out what's going on, I have to disable ```raise_errors``` on the Client object.

```ruby
access_token.client.options[:raise_errors] = false
```

Then I can get a Response object that shows the http status. It would be helpful if the OAuth2 error contained the http status code and text, at the bare minimum... Especially when the body is empty.

Thanks!

**Comment by pboling** [comment]
Hi @ginjo !

If possible, please try try pointing your Gemfile to master, and see how that works out.

```
gem 'oauth', git: 'https://github.com/oauth-xx/oauth2.gi', branch: 'master'
```

The way errors are handled has changed a bit.  I don't think this is a bug we will fix on the 1.4.x line, so it will be fixed, and may have already been fixed, in 2.0.0, which is not quite ready for release.

Please test your app against master!

**Comment by ginjo** [comment]
Thanks @pboling , I'll give that a try. I'll have to do some mocking, as the bug in Slack's API that caused an empty body for some responses has since been addressed. I'll look through the OAuth2 specs & tests for a good place to jump in.
